### PR TITLE
[Teams] Office365 Connector Card Scenario

### PR DESCRIPTION
**Pull request by tsuwandy**
Fixes https://github.com/microsoft/botbuilder-js/issues/1202

## Description
Teams Office 365 Connector Bot

### CODE REVIEW

File: libraries/botbuilder/tests/teams/office365Card/src/office365CardsBot.ts [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+//
+// You can install this bot in any scope. From the UI just @mention the bot.
+//
+export class Office365CardsBot extends TeamsActivityHandler {
+    constructor() {
+        super();
+
+        // See https://aka.ms/about-bot-activity-message to learn more about the message and other activity types.
+        this.onMessage(async (context, next) => {
+            console.log("ONMESSAGE");
```
[the remark this answers is not in the document]
Comment (tsuwandy) [reply libraries/botbuilder/tests/teams/office365Card/src/office365CardsBot.ts]: ooops, nice catch Steven!!

File: libraries/botbuilder/tests/teams/office365Card/src/office365CardsBot.ts [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+        });
+    }
+
+    protected async onTeamsO365ConnectorCardAction(context: TurnContext, query: O365ConnectorCardActionQuery): Promise<void> {
+        await context.sendActivity(MessageFactory.text(`O365ConnectorCardActionQuery event value: ${JSON.stringify(query)}`));
+    }
+
+    private async sendO365CardAttachment(context: TurnContext): Promise<void> {
+        const card = CardFactory.o365ConnectorCard({
+            "title": "card title",
```
Comment (daveta) [inline libraries/botbuilder/tests/teams/office365Card/src/office365CardsBot.ts]: Optional: I was adding casts so at least you can refer back to the underlying interfaces. It gives some level of "documentation".

File: libraries/botbuilder/tests/teams/office365Card/src/office365CardsBot.ts [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+                    "os": "windows",
+                    "uri": "http://microsoft.com"
+                    }
+                ],
+                "name": "Open Uri",
+                "@id": "open-uri"
+                }
+            ]
+        });
+        console.log(JSON.stringify(card));
```
Comment (daveta) [inline libraries/botbuilder/tests/teams/office365Card/src/office365CardsBot.ts]: >console.log [](start = 8, length = 11)

Not sure if this was intentional.

File: libraries/botbuilder/tests/teams/office365Card/src/office365CardsBot.ts (hunk 2 above)
Comment (stevengum) [reply libraries/botbuilder/tests/teams/office365Card/src/office365CardsBot.ts]: @tsuwandy, I checked the botbuilder-core exports and it exports the O365Connector interfaces:

o364 stuff

File: libraries/botbuilder/tests/teams/office365Card/src/office365CardsBot.ts (hunk 3 above)
Comment (tsuwandy) [reply libraries/botbuilder/tests/teams/office365Card/src/office365CardsBot.ts]: doh! thanks dave!

File: libraries/botbuilder/tests/teams/office365Card/src/office365CardsBot.ts (hunk 2 above)
Comment (tsuwandy) [reply libraries/botbuilder/tests/teams/office365Card/src/office365CardsBot.ts]: Hmm, the schema seems to be a mismatch with the json we are generating from dotnet sample is not compatible.  Do we need to regenerate them?

**Pull request merged**